### DISCUSSION TITLE
Integration with diffview

### DISCUSSION THREAD

**Discussion by ericsunplus** [NeogitOrg/neogit#1616] [Q&A] [answered]
Hi community,

I want the neogit diff show as `diffview` style. In the README.md. It indicates a configuration
```lua
neogit.setup {
  integrations = {
    ...
    -- Neogit only provides inline diffs. If you want a more traditional way to look at diffs, you can use `diffview`.
    -- The diffview integration enables the diff popup.
    --
    -- Requires you to have `sindrets/diffview.nvim` installed.
    diffview = nil,
    ...
}
```

However after I set this, my diff is still shown as a inline diff way instead of a diffview side by side style. I open a `NeogitLogView` window first then press `<CR>` on one of the commit, this is what I get
![image](https://github.com/user-attachments/assets/8d42a93e-6c89-44c4-b24f-ae322619dd58)

I can achieve what I want by `:DiffviewOpen <commit>^!`. But this behavior is not "integrated" to neogit. Can anyone point out what may be missing in my configuration?

My full neogit config is as the following
```lua
return {
    "NeogitOrg/neogit",
    dependencies = {
        "nvim-lua/plenary.nvim",
        "sindrets/diffview.nvim",
    },
    config = function()
        require('neogit').setup {
            integrations = {
                diffview = true
            }
        } 
        vim.keymap.set('n', '<Leader>g', ':Neogit<CR>')
        vim.keymap.set('n', '<Leader>gh', ':DiffviewFileHistory ')
    end
}
```

**Comment by CKolkey** [comment, marked as answer]
Ah, perhaps the issue here is on _how_ it's integrated. The diffview integration is for diffing via the diff popup (`d`). The status buffer (the view you are asking about) only offers inline diffs, not side-by-side.

This makes staging/unstaging/discarding individual lines possible, which is handy.

**Reply by ericsunplus** [reply]
Hi @CKolkey , thank you for the quick response. I think I would have to made a customized mapping to trigger this

```lua
        vim.keymap.set("n", "<Leader>dv", function()
            local sha = vim.fn.expand("<cword>")
            print("DiffviewOpen " .. sha .. "^!")
            vim.cmd("DiffviewOpen " .. sha .. "^!")
        end)
```

Is the `NeogitLogView` a status buffer? I see in the README.md mentioning about "log_view". Should I add such mapping to "status" or "log_view"?

**Reply by CKolkey** [reply]
"log_view" is what is opened via the log popup (so `ll` `lc`, etc). the status buffer is what is visible with `:Neogit`

**Reply by ericsunplus** [reply]
OK. I am just add a global mapping and it works perfectly. Thank you for the great plugin

**Comment by A7Lavinraj** [comment]
@CKolkey I want to enable single buffer diff view instead of side by side view
